funcx_endpoint/tests/unit/test_cli_behavior.py

```diff
@@ -45,36 +49,69 @@ def cli_runner():
 def run_line(cli_runner):
     def func(argline, *, assert_exit_code: int | None = 0):
         args = shlex.split(argline) if isinstance(argline, str) else argline
+
         result = cli_runner.invoke(app, args)
         if assert_exit_code is not None:
-            assert result.exit_code == assert_exit_code
+            assert result.exit_code == assert_exit_code, (result.stdout, result.stderr)
         return result
 
     return func
 
 
-def test_start_endpoint_no_such_ep(run_line, endpoint_obj):
-    res = run_line("start foo")
-    endpoint_obj.start_endpoint.assert_not_called()
-    assert "Endpoint foo is not configured" in res.stdout
+def test_start_ep_corrupt(run_line, mock_cli_state, make_endpoint_dir):
+    make_endpoint_dir("foo")
+    mock_ep, mock_state = mock_cli_state
+    conf = mock_state.endpoint_config_dir / "foo" / "config.py"
+    conf.unlink()
+    res = run_line("start foo", assert_exit_code=1)
+    assert "corrupted?" in res.stderr
+
+
+def test_start_endpoint_no_such_ep(run_line, mock_cli_state):
+    res = run_line("start foo", assert_exit_code=1)
+    mock_ep, _ = mock_cli_state
+    mock_ep.start_endpoint.assert_not_called()
+    assert "Endpoint 'foo' is not configured" in res.stderr
 
 
-def test_start_endpoint_existing_ep(run_line, endpoint_obj, make_endpoint_dir):
+def test_start_endpoint_existing_ep(run_line, mock_cli_state, make_endpoint_dir):
     make_endpoint_dir("foo")
     run_line("start foo")
-    endpoint_obj.start_endpoint.assert_called_once()
+    mock_ep, _ = mock_cli_state
+    mock_ep.start_endpoint.assert_called_once()
 
 
-def test_stop_endpoint(run_line, endpoint_obj, make_endpoint_dir):
+def test_stop_endpoint(run_line, mock_cli_state, make_endpoint_dir):
     run_line("stop foo")
-    endpoint_obj.stop_endpoint.assert_called_once()
+    mock_ep, _ = mock_cli_state
+    mock_ep.stop_endpoint.assert_called_once()
 
 
 def test_restart_endpoint_does_start_and_stop(
-    run_line, endpoint_obj, make_endpoint_dir
+    run_line, mock_cli_state, make_endpoint_dir
 ):
     make_endpoint_dir("foo")
     run_line("restart foo")
 
-    endpoint_obj.stop_endpoint.assert_called_once()
-    endpoint_obj.start_endpoint.assert_called_once()
+    mock_ep, _ = mock_cli_state
+    mock_ep.stop_endpoint.assert_called_once()
+    mock_ep.start_endpoint.assert_called_once()
+
+
+def test_start_ep_incorrect(run_line, mock_cli_state, make_endpoint_dir):
```

Comment:
Like the plethora of tests.  Probably because you ran into these while manually testing?

Reply:
Bing. Go.  Ugh.